factory/animal.h

```diff
@@ -0,0 +1,54 @@
+
+#ifndef __ANIMAL_PLANET_ANIMAL_HPP_
+#define __ANIMAL_PLANET_ANIMAL_HPP_
+
+#include <memory>
+#include <string>
+
+namespace animal
+{
+
+//using AnimalPtr = std::unique_ptr<CAnimal>;
+class CAnimal
+{
+public:
+    CAnimal(std::string sName = {}, std::string sType = "");
+
+    virtual std::string GetType() const = 0;
+
+    virtual void SetName(std::string sName) = 0;
+    virtual std::string GetName() const = 0;
+
+protected:
+    std::string m_sName{};
```

Comment:
Generally you don't need to call the default constructor `{}` for string initialization. 
It automatically calls default ctor, the value is "" (empty string)